go/src/koding/klient/app/defertime.go

```diff
@@ -0,0 +1,82 @@
+package app
+
+import "time"
+
+// DeferTime allows to invoke registered function after provided duration. It
+// can cancel deferred function call even if internal timer was already started.
+type DeferTime struct {
```

Comment:
@rjeczalik I'm thinking about long running go-routines like [this](https://github.com/koding/koding/blob/ed922385cd322370f31e2fe8c97a9da89dce1146/go/src/koding/klient/app/klient.go#L577) they are also "leaked" but this is fine, since klient object is meant to act like a singleton.